### ISSUE TITLE
commit "handle modules comments"

### ISSUE THREAD

**Issue by udeved** [OpenRC/openrc#95]
This commit produces actually messages at boot, attempting to load a comment.
I have this on 3 different machines.

https://github.com/OpenRC/openrc/commit/c289774b00d0d7dc38fdc1f0f623569bd184a4b1

~~~
 * Loading module ## ...
 * Failed to load ##
 [ !! ]
 * Loading module ## ...
 * Failed to load ##
 [ !! ]
 * Loading module ## ...
 * Failed to load ##
 [ !! ]
 * Loading module nvidia ...
 * Failed to load nvidia
 [ !! ]
 * Loading module vboxdrv ...
 * Failed to load vboxdrv
 [ !! ]
 * Loading module vboxpci ...
 * Failed to load vboxpci
 [ !! ]
 * Loading module vboxnetadp ...
 * Failed to load vboxnetadp
 [ !! ]
 * Loading module vboxnetflt ...
 * Failed to load vboxnetflt
 [ !! ]
 * Loading module nvidia ...
 [ ok ]
 * Loading module vboxdrv ...
 [ ok ]
 * Loading module vboxnetadp ...
 [ ok ]
 * Loading module vboxpci ...
 [ ok ]
 * Loading module vboxnetflt ...
 [ ok ]
~~~

**Comment by dwfreed** [comment]
It would be helpful if you provided the contents of your /usr/lib/modules-load.d/*, /run/modules-load.d/*, and /etc/modules-load.d/*, and their paths (this is important due to the override behavior).

**Comment by udeved** [comment]
Here you go:

~~~
$ ls /etc/modules-load.d
mhwd-gpu.conf  modules.conf
~~~

~~~
ls /usr/lib/modules-load.d/
virtualbox-host-dkms.conf
~~~

/run/modules-load.d doesn't exist on arch/manjaro.

~~~
cat /etc/modules-load.d/modules.conf
# List of modules to load at boot
~~~

~~~
cat /etc/modules-load.d/mhwd-gpu.conf
##
## Generated by mhwd - Manjaro Hardware Detection
##
 
nvidia
~~~

~~~
$ cat /usr/lib/modules-load.d/virtualbox-host-dkms.conf
vboxdrv
vboxpci
vboxnetadp
vboxnetflt
~~~

I just realized, it may be caused by the patch I add?
https://github.com/manjaro/packages-openrc/blob/master/openrc/use-optional-modules-load-d.patch

**Comment by dwfreed** [comment]
Is this happening in `/etc/init.d/modules` or `/etc/init.d/modules-load`? (Zap them both, then try to start them again, modules-load first)

**Comment by udeved** [comment]
The patch add optional modules-load.d support, but it is set false by default.
The boot msg are produced with standard /etc/conf.d/modules set.

~~~
$ cat /etc/conf.d/modules
# You can define a list modules for a specific kernel version,
# a released kernel version, a main kernel version or just a list.
# The most specific versioned variable will take precedence.

# if openrc should make use of /etc/modules-load.d
# systemd places stuff there
use_modules_load_d=false

#modules_2_6_23_gentoo_r5="ieee1394 ohci1394"
#modules_2_6_23="tun ieee1394"
#modules_2_6="tun"
#modules_2="ipv6"
#modules="ohci1394"

# You can give modules a different name when they load - the new name
# will also be used to pick arguments below.
#modules="dummy:dummy1"

# Give the modules some arguments if needed, per version if necessary.
# Again, the most specific versioned variable will take precedence.
#module_ieee1394_args="debug"
#module_ieee1394_args_2_6_23_gentoo_r5="debug2"
#module_ieee1394_args_2_6_23="debug3"
#module_ieee1394_args_2_6="debug4"
#module_ieee1394_args_2="debug5"

# You should consult your kernel documentation and configuration
# for a list of modules and their options.

modules="nvidia vboxdrv vboxnetadp vboxpci vboxnetflt"
~~~

**Comment by dwfreed** [comment]
You didn't answer my question.  Does this happen when starting /etc/init.d/modules-load, or /etc/init.d/modules? As I said, zap them both, then try starting modules-load first (modules has 'want modules-load', so starting modules will start modules-load first).

**Comment by udeved** [comment]
With the new modules-load service started dynamically.

~~~
Dynamic Runlevel: needed/wanted
 modules-load                                                                                          [  started  ]
 xdm-setup                                                                                             [  started  ]
~~~

**Comment by dwfreed** [comment]
You *don't* have commit c289774 applied.  The comments are ignored properly with that commit applied.  As for why the other loads are failing, try removing the -q from the modload command to see what error it gives.

**Comment by udeved** [comment]
I built from 0.21.4 release, no git version here.
I think its too early to close.

**Comment by dwfreed** [comment]
c289774 isn't in 0.21.4; I'm waiting for @williamh to make a release.

**Comment by udeved** [comment]
Aha, ok, so 0.21.5 will fix it, fine with me, since its only messages that can be ignored and modules still load.

**Comment by udeved** [comment]
Nope, 0.21.5 still produces error messages at boot.
I removed the patch to make sure, and used default /etc/conf.d/modules

```
* Loading module nvidia ...
 * Failed to load nvidia
 [ !! ]
 * Loading module vboxdrv ...
 * Failed to load vboxdrv
 [ !! ]
 * Loading module vboxpci ...
 * Failed to load vboxpci
 [ !! ]
 * Loading module vboxnetadp ...
 * Failed to load vboxnetadp
 [ !! ]
 * Loading module vboxnetflt ...
 * Failed to load vboxnetflt
 [ !! ]
 * Autoloaded 0 module(s)
```

**Comment by udeved** [comment]
PR #96 fixes it in 0.21.5
